syntax = "proto3";

package hr.domain;

message Experience {
    string companyName = 1;
    string title = 2;
    string datesEmployedText = 3;
    string startDateYYYYMMDD = 4;
    string employmentDurationText = 5;
    int32 employmentDurationInMonths = 6;
    string location = 7;
    string experienceText = 8;

}

message Education {
    string schoolName = 1;
    string degreeName = 2;
    string startDateYYYY = 3;
    string endDateYYYY = 4;

}


message Resume {
    string name = 1;
    string rawResume = 2;
    string profileURL = 3;
    float monthlySalary = 4;
    string companyName = 5; // from role
    string title = 6; // from role
    string aboutRaw = 7;
    string experienceRaw = 8;
    string educationRaw = 9;
    string licensesCertificationsRaw = 10;
    string skillsEndorsementsRaw = 11;

    repeated Experience experiences = 16;
    repeated Education educations = 17;


}

message ResumeDB {
    repeated Resume resumes = 1;
}